### ISSUE TITLE
Issue with User Form

### ISSUE THREAD

**Issue by akash65** [osTicket/osTicket#4584]
Hello everyone,

While creating ticket on user dashboard, i created manual form to fill details what i needed is i want to hide some fields from agent not to show after the ticket assigned to agent? is that possible in osticket

I have attached screenshots below

![screenshot 221](https://user-images.githubusercontent.com/12798776/47723023-af991980-dc79-11e8-8eb8-8aee91e34ec4.png)
![screenshot 222](https://user-images.githubusercontent.com/12798776/47723035-b45dcd80-dc79-11e8-990c-c41c66c4a93a.png)

**Comment by JediKev** [comment]
@akash65 

By "hide some fields from agent not to show after the ticket assigned to agent" you mean hide the field from USERS but show field to AGENTS. There is a _huge_ difference between Users and Agents 😄 

Simply go to the field config (**Admin Panel > Manage > Forms > click form > find field > Config**), make the field non-visible/non-editable to **EndUsers**, and **Save Changes**. This will hide the field from the User portal. This will also hide the field from User ticket create though so they won't be able to insert data when creating a ticket. There is no way currently in the system to hide a field on ticket view but not ticket create; you have to hide the field on ticket create and view or not at all.

Cheers.

**Comment by akash65** [comment]
No, i want to hide custom form fields from AGENT once ticket get assigned and it will be visible to ENDUSER but not AGENT header. is that possible?

![screenshot 223](https://user-images.githubusercontent.com/12798776/47767405-8029f180-dcf9-11e8-82af-eaf3d0868b42.png)
![screenshot 224](https://user-images.githubusercontent.com/12798776/47767411-861fd280-dcf9-11e8-9ab6-389cd036b13b.png)
![screenshot 225](https://user-images.githubusercontent.com/12798776/47767419-8d46e080-dcf9-11e8-9cca-c7c9557bb28d.png)
![screenshot 226](https://user-images.githubusercontent.com/12798776/47767425-92a42b00-dcf9-11e8-8331-b80a101c30de.png)
![screenshot 227](https://user-images.githubusercontent.com/12798776/47767431-989a0c00-dcf9-11e8-9895-65ee75490f90.png)

**Comment by Acu2Ind** [comment]
@JediKev : Hope the sequential view of the pictures above made the issue clear. 
Summarizing below just to get the wording right:

 1)PO Amount is a new field created in a custom form, with configuration as "Visible: For End Users Only and Editable: For End Users Only"

2)When the End User creates a new ticket, the PO Amount field is visible, wherein he can enter the value.

3)Upon submission of the ticket, it gets routed to an Agent of a Dept (as configured in the system, using Help Topic)

4)Issue is the Agent is ALSO able to see the field value (PO Amount=26867 in this case) - even though field configuration allowed ONLY the End User  to see or edit it.

Please help!!

**Comment by Acu2Ind** [comment]
For general visibility on OsTicket Version: v1.10.1

**Comment by JediKev** [comment]
@Acu2Ind 

You'd have to write some custom code to hide the field AFTER assignment.

But if you want to hide the field from Agents then change the field to be non-visible and non-editable to the Agents.

Cheers.

**Comment by Acu2Ind** [comment]
Thanks for your response, JK.. No such idea of hiding the field AFTER assignment.

We want to implement the latter scenario you mentioned i.e hide the field from Agents, and we're trying to do so by making it non-visible and non-editable to them (and visible and editable only to End User). However, it simply does NOT work. The Agent is still able to see it.  Refer the images above.

Need help!

**Comment by Acu2Ind** [comment]
@JediKev : Tagging for SOS!!

**Comment by JediKev** [comment]
@Acu2Ind 

Was that ticket created before you changed the permissions? Create a new ticket and see the results.

Cheers.

**Comment by Acu2Ind** [comment]
No. The tickets were created only AFTER the field was configured to be visible and editable to ONLY the End Users. Of course, it was all saved. The EndUser was logged out for extra caution. And then logged back in to create the ticket. The reported issue persists. 

We believe it's a bug. Please let us know if you again need the screenshots.

**Comment by JediKev** [comment]
@Acu2Ind 

Just got word from another dev here and they say we have issues with field permissions. Will be addressed in `1.11-RC2` coming soon.

Cheers.

**Comment by Acu2Ind** [comment]
Oh..Ok. Thanks for acknowledging.

Because we need this particular issue rectified urgently, any guidance on self help? Ex. PHP file/line no. - Thanks in advance :)

**Comment by nicoletta-maia** [comment]
Hi @Acu2Ind  
you could add a new filter in the query to line _446_ in `include/staff/ticket-view.inc.php`. Something similar to:
```
$answers = $form->getAnswers()->exclude(Q::any(array(
        'field__flags__hasbit' => DynamicFormField::FLAG_EXT_STORED,
        'field__name__in' => array('subject', 'priority')
    )))
    ->filter(array(
        'field__flags__hasbit' => DynamicFormField::FLAG_AGENT_VIEW,
    ));
```
I suggest you also update `include/staff/templates/ticket-print.tmpl.php` to prevent agents from seeing the fields by printing the ticket.

Hope it helped

**Comment by akash65** [comment]
Thank you so much @nicoletta-maia .It works
